### PR TITLE
feat: increment consensus ver and register migration

### PR DESCRIPTION
**Pull request by smarshall-spitzbart**
## Description
 
The consumer consensus version was not incremented in https://github.com/cosmos/interchain-security/pull/1037 when it should have been. Also `MigrateConsumerPacketData` was not registered in 1037 when it should have been. This PR fixes both issues. 

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] Added `!` to the type prefix if state-machine breaking change (i.e., requires coordinated upgrade)
* [x] Confirmed this PR does not introduce changes requiring state migrations, OR migration code has been added to consumer and/or provider modules
* [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
* [x] Provided a link to the relevant issue or specification
* [x] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules)
* [x] Included the necessary unit and integration [tests](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#testing)
* [x] Added a changelog entry to `CHANGELOG.md`
* [x] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [x] Updated the relevant documentation or specification
* [x] Reviewed "Files changed" and left comments if necessary 
* [ ] Confirmed all CI checks have passed
* [x] If this PR is library API breaking, bump the go.mod version string of the repo, and follow through on a new major release for both the consumer and provider

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] confirmed this PR does not introduce changes requiring state migrations, OR confirmed migration code has been added to consumer and/or provider modules
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

File: x/ccv/consumer/module.go [hunk 1]
```diff
@@ -107,6 +107,9 @@ func (AppModule) RegisterInvariants(ir sdk.InvariantRegistry) {
 // RegisterServices registers module services.
 func (am AppModule) RegisterServices(cfg module.Configurator) {
 	consumertypes.RegisterQueryServer(cfg.QueryServer(), am.keeper)
+	cfg.RegisterMigration(consumertypes.ModuleName,
+		1, // from version 1
+		am.keeper.MigrateConsumerPacketData)
```
Comment (mpoke) [inline x/ccv/consumer/module.go:112]: Use SDK model https://github.com/cosmos/cosmos-sdk/blob/main/x/staking/keeper/migrations.go

Comment (smarshall-spitzbart) [reply x/ccv/consumer/module.go:112]: You meant just pattern matching their top level migration funcs? https://github.com/cosmos/interchain-security/commit/a21eefcbb25285bdadf59952b9637a7c10e79243

Comment (mpoke) [reply x/ccv/consumer/module.go:112]: Yeah, but then you need to register Migrate1to2 here.

Comment (smarshall-spitzbart) [reply x/ccv/consumer/module.go]: https://github.com/cosmos/interchain-security/commit/0aa82001c6cf3945f58f34b07500e51e1595a14a

**Review by mpoke — APPROVED** [review]
LGTM

**Comment by smarshall-spitzbart** [comment]
#1297 will need to be merged before this PR

**Review by p-offtermatt — APPROVED** [review]
LGTM, but I don't have experience with migrations. Mostly want to approve so you can merge if it becomes blocking, but if anyone else wants to review, feel free.

**Pull request merged**